### PR TITLE
Windows XP support test

### PR DESCRIPTION
**Pull request by xarkes**

### CODE REVIEW

**Comment by xarkes** [comment]
There are still some linking issues (QueryFullProcessImageNameA not found in Kernel32.dll)

**Comment by radare** [comment]
Test or ready to merge?

**Comment by xarkes** [comment]
Not yet I want to fix the linking issues first

**Comment by radare** [comment]
let me know when ready

**Comment by xarkes** [comment]
Ok the binary starts under windows XP however there are some bugs (path related issues)

**Comment by xarkes** [comment]
Meh I am not very satisfied there are still a few warnings under windows XP, but otherwise it looks good to me. I don't know if you want to merge before release @pancake, I did not test everything.

File: libr/util/file.c [hunk 1]
```diff
@@ -864,9 +864,18 @@ R_API int r_file_mkstemp(const char *prefix, char **oname) {
 
 R_API char *r_file_tmpdir() {
 #if __WINDOWS__
-	char *path = r_sys_getenv ("TEMP");
-	if (!path) {
-		path = strdup ("C:\\WINDOWS\\Temp\\");
+	TCHAR tmpdir[260];
```
Comment (radare) [inline libr/util/file.c]: Use maxpathlenth define instead of 260 magic

File: libr/util/file.c [hunk 2]
```diff
@@ -864,9 +864,18 @@ R_API int r_file_mkstemp(const char *prefix, char **oname) {
...
+	char *path = NULL;
+
```
Comment (radare) [inline libr/util/file.c]: Use sizeof instead of 260

File: libr/util/sys.c [hunk 3]
```diff
@@ -877,6 +880,24 @@ R_API char *r_sys_pid_to_path(int pid) {
...
+	if (processHandle != NULL) {
+		if (GetModuleFileNameEx (processHandle, NULL, filename, FILENAME_MAX) == 0) {
+			eprintf ("r_sys_pid_to_path: Cannot get module filename.");
+		} else {
+			return strdup (filename);
+		}
+		CloseHandle (processHandle);
+	} else {
+		eprintf ("r_sys_pid_to_path: Cannot open process.");
+	}
```
Comment (radare) [inline libr/util/sys.c]: Empty line

File: sys/meson_extra.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# meson sucks a bit for now and they are slow on taking pull requests so
+# this is a hack to do what I want
```
Comment (radare) [inline sys/meson_extra.py]: Can we have a hack like this for sdb?

**Comment by radare** [comment]
See comments and ill merge

File: sys/meson_extra.py (hunk 4 above)
Comment (xarkes) [inline sys/meson_extra.py]: What do you mean ? To compile sdb on appveyor with xp support ?

**Comment by radare** [comment]
I mean to modify meson to handle the precompilation of sdb files. But i also think that sdb should have an api to load a plaintext k=v file

**Comment by XVilka** [comment]
May be merge this before the release? So people can test it on Windows XP - just add in the release notes that support is "experimental".

**Pull request merged**